coreclient/src/mimi_content/mod.rs

```diff
@@ -1,274 +0,0 @@
-    url: ExternalPartUrl,       // A URL where the content can be fetched
-    expires: Option<TimeStamp>, // This is actually a u32 and needs to be parsed as such. 0 means no expiration, i.e. None.
-    size: u64,                  // size of content in octets
-    aead_alg: AeadAlg,          // An IANA AEAD Algorithm number, or zero
-    // TODO: Key and nonce need their own types.
-    key: Vec<u8>,             // AEAD key
-    nonce: Vec<u8>,           // AEAD nonce
-    aad: Vec<u8>,             // AEAD additional authentiation data
-    description: TlsStrOwned, // an optional text description
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-struct MultiParts {
-    pars: Vec<NestablePart>,
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-#[repr(u16)]
-enum PartSemantics {
-    NullPart,
-    SinglePart,
-    ChooseOne,
-    SingleUnit,
-    ProcessAll,
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-#[repr(u16)]
-enum Part {
-    Null,
-    Single(SinglePart),
-    External(ExternalPart),
-    Multi(MultiParts),
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-#[repr(u16)]
-enum Disposition {
-    Unspecified,
-    Render,
-    Reaction,
-    Profile,
-    Inline,
-    Icon,
-    Attachment,
-    Session,
-    Preview,
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-struct Language {
-    language: TlsStrOwned,
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-struct NestablePart {
-    disposition: Disposition,
-    languages: Vec<Language>,
-    part_index: u16,
-    part_semantic: PartSemantics,
-    part: Part,
-}
-
-impl Default for NestablePart {
-    fn default() -> Self {
-        Self {
-            disposition: Disposition::Unspecified,
-            languages: Vec::new(),
-            part_index: 0,
-            part_semantic: PartSemantics::NullPart,
-            part: Part::Null,
-        }
-    }
-}
-
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-struct MessageDerivedValues {
-    mls_group_id: GroupId,
-    sender_leaf_index: u32,
-    sender_client_id: AsClientId,
-    sender_user_id: QualifiedUserName,
-    group_name: TlsStrOwned,
-}
-
-// WARNING: If this type is changed, the storage and loading logic in the
-// `crate::conversations::messages::peristence` module must be updated
-// accordingly and the `MESSAGE_CONTENT_FORMAT_VERSION` constant must be
-// incremented by one.
-#[derive(
-    PartialEq, Debug, Clone, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserializeBytes,
-)]
-pub struct MimiContent {
-    pub id: MessageId,
```

Comment:
Messages are not aware of their message id. Did you plan to use this for anything? We still have an id in ConversationMessage::conversation_id